protocol/synthetix/contracts/storage/AccountDelegationIntents.sol

```diff
@@ -0,0 +1,170 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.11 <0.9.0;
+
+import "./DelegationIntent.sol";
+import "@synthetixio/core-contracts/contracts/utils/SetUtil.sol";
+import "@synthetixio/core-contracts/contracts/utils/SafeCast.sol";
+import "../interfaces/IVaultModule.sol";
+import "./Account.sol";
+
+/**
+ * @title Represents a delegation (or undelegation) intent.
+ */
+library AccountDelegationIntents {
+    using SafeCastI256 for int256;
+    using SafeCastU128 for uint128;
+    using SafeCastU256 for uint256;
+    using SetUtil for SetUtil.UintSet;
+    using SetUtil for SetUtil.AddressSet;
+    using DelegationIntent for DelegationIntent.Data;
+    using Account for Account.Data;
+
+    struct Data {
+        uint128 accountId;
+        uint128 delegationIntentsEpoch; // nonce used to nuke previous intents using a new era (useful on liquidations)
+        SetUtil.UintSet intentsId;
+        mapping(bytes32 => SetUtil.UintSet) intentsByPair; // poolId/collateralType => intentIds[]
+        // accounting for the intents collateral delegated
+        // Per Collateral
+        SetUtil.AddressSet delegatedCollaterals;
+        mapping(address => int256) netDelegatedAmountPerCollateral; // collateralType => net delegatedCollateralAmount
+    }
+
+    /**
+     * @dev Returns the account delegation intents stored at the specified account id.
+     */
+    function load(
+        uint128 accountId,
+        uint128 delegationIntentsEpoch
+    ) internal pure returns (Data storage accountDelegationIntents) {
+        bytes32 s = keccak256(
+            abi.encode(
+                "io.synthetix.synthetix.AccountDelegationIntents",
+                accountId,
+                delegationIntentsEpoch
+            )
+        );
+        assembly {
+            accountDelegationIntents.slot := s
+        }
+    }
+
+    /**
+     * @dev Returns the account delegation intents stored at the specified account id.
+     */
+    function loadValid(
+        uint128 accountId
+    ) internal view returns (Data storage accountDelegationIntents) {
+        uint128 delegationIntentsEpoch = Account.load(accountId).currentDelegationIntentsEpoch;
+        accountDelegationIntents = load(accountId, delegationIntentsEpoch);
+        if (
+            accountDelegationIntents.accountId != 0 &&
+            (accountDelegationIntents.accountId != accountId ||
+                accountDelegationIntents.delegationIntentsEpoch != delegationIntentsEpoch)
+        ) {
+            revert IVaultModule.InvalidDelegationIntent();
+        }
+    }
+
+    /**
+     * @dev Returns the account delegation intents stored at the specified account id. Checks if it's valid
+     */
+    function getValid(uint128 accountId) internal returns (Data storage accountDelegationIntents) {
```

Comment:
the name of this function and understanding the difference between `getValid` and `loadValid` seems extremely difficult. maybe a better name is `loadWithInit` or something?